FieldOpt/Hdf5SummaryReader/hdf5_summary_reader.h

```diff
@@ -249,16 +293,38 @@ class Hdf5SummaryReader {
     };
 
     void readTimeVector(std::string file_path); //!< Read the time vector from the HDF5 summary file.
-    void readActiveCells(std::string file_path); //!< Read vector defining which cells are active from the HDF5 summary file.
     void readWellStates(std::string file_path); //!< Read all well state information from the HDF5 summary file.
     void parseWsVector(std::vector<wstype_t> &wsvec); //!< Populate well_states_ by creating well_data and perforation_data objects from the wstype_t vector.
     well_data parseWellState(std::vector<wstype_t> &ws, int wnr); //!< Parse the states for a single well and create a well_data object.
 
+    void readActiveCells(std::string file_path); //!< Read vector defining which cells are active from the HDF5 summary file.
+    void readReservoirPressure(std::string file_path); //!< Read reservoir cell pressures for each time step.
+    void readSaturation(std::string file_path); //!< Read cell saturations for each time step.
+
+    /*!
+     * variables containing information about the reservoir cell ensemble,
+     * e.g., number of active cells, corresponding active cell indices, etc.
+     */
+    std::vector<int> cells_all_vector_; //!< Vector def. status for all cells (size equal to total number of cells).
```

Comment:
Very much agree. Started actually doing this, but had some newbie problems, and returned to this setup. Leaving it as todo work for now.